ui/desktop/src/components/settings_v2/extensions/utils.ts

```diff
@@ -166,26 +166,18 @@ export async function replaceWithShims(cmd: string) {
   return cmd;
 }
 
-export function removeShims(cmd: string) {
-  // Only remove shims if the path matches our known shim patterns
-  const shimPatterns = [
-    /\/goose-shims\/goosed$/,
-    /\/goose-shims\/jbang$/,
-    /\/goose-shims\/npx$/,
-    /\/goose-shims\/uvx$/,
-  ];
-
-  // Check if the command matches any shim pattern
-  const isShim = shimPatterns.some((pattern) => pattern.test(cmd));
-
-  if (isShim) {
-    const segments = cmd.split('/');
-    // Filter out any empty segments (which can happen with trailing slashes)
-    const nonEmptySegments = segments.filter((segment) => segment.length > 0);
-    // Return the last segment or empty string if there are no segments
-    return nonEmptySegments.length > 0 ? nonEmptySegments[nonEmptySegments.length - 1] : '';
+export async function removeShims(cmd: string) {
```

Comment:
if this is now `async` we'll need to `await` where it's called

Reply:
yup... this is a long chain of async's/awaits

Reply:
can we force this function to resolve the promise synchronously so we don't need to make changes across the board?